api/src/app.py

```diff
@@ -156,6 +157,11 @@ def register_blueprints(app: APIFlask) -> None:
     app.register_blueprint(task_blueprint)
     app.register_blueprint(load_search_data_blueprint)
 
+    # Only enable in lower envs until https://github.com/HHS/simpler-grants-gov/issues/4345
+    # has been resolved.
+    if not endpoint_config.is_prod:
+        app.register_blueprint(legacy_soap_api_blueprint)
```

Comment:
Don't setup environment behavior like this - instead add a boolean that turns it on/off by environment like we do for the other configurations.